app/services/registration_checker.rb

```diff
@@ -89,6 +91,20 @@ def validate_create_events!
       raise RegistrationError.new(:forbidden, ErrorCodes::INVALID_EVENT_SELECTION) if event_limit.present? && event_ids.count > event_limit
     end
 
+    def validate_qualifications!
+      return unless @competition_info.enforces_qualifications?
+      # TODO: Read the request payload in as an object, and handle cases where expected values aren't found
+      event_ids = @request.dig('competing', 'event_ids')
+
+      unqualified_event = event_ids.find do |event|
+        qualification = @competition_info.get_qualification_for(event)
+        next if qualification.nil?
```

Comment:
Using `next` inside `find` (or any functional call like `map` etc.) is highly unorthodox.
The purpose of `find` is to skip the entry "on your behalf" so long as you return `false` from the block. So something like `qualification.present? && competitor_qualifies` as a return value should do, eliminating the need for this one line.

Reply:
Addressed